exporter/awsemfexporter/metric_translator.go

```diff
@@ -77,75 +58,23 @@ type CWMetricStats struct {
 	Sum   float64
 }
 
-// Wrapper interface for:
-//  - pdata.IntDataPointSlice
-//  - pdata.DoubleDataPointSlice
-//  - pdata.IntHistogramDataPointSlice
-//  - pdata.DoubleHistogramDataPointSlice
-//  - pdata.DoubleSummaryDataPointSlice
-type DataPoints interface {
-	Len() int
-	At(int) DataPoint
-}
-
-// DataPoint is a wrapper interface for:
-//  - pdata.IntDataPoint
-//  - pdata.DoubleDataPoint
-//  - pdata.IntHistogramDataPoint
-//  - pdata.DoubleHistogramDataPoint
-//  - pdata.DoubleSummaryDataPointSlice
-type DataPoint interface {
-	LabelsMap() pdata.StringMap
-}
-
-// Define wrapper interfaces such that At(i) returns a `DataPoint`
-type IntDataPointSlice struct {
-	pdata.IntDataPointSlice
-}
-type DoubleDataPointSlice struct {
-	pdata.DoubleDataPointSlice
-}
-type DoubleHistogramDataPointSlice struct {
-	pdata.DoubleHistogramDataPointSlice
-}
-type DoubleSummaryDataPointSlice struct {
-	pdata.DoubleSummaryDataPointSlice
-}
-
-func (dps IntDataPointSlice) At(i int) DataPoint {
-	return dps.IntDataPointSlice.At(i)
-}
-func (dps DoubleDataPointSlice) At(i int) DataPoint {
-	return dps.DoubleDataPointSlice.At(i)
-}
-func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
-	return dps.DoubleHistogramDataPointSlice.At(i)
-}
-func (dps DoubleSummaryDataPointSlice) At(i int) DataPoint {
-	return dps.DoubleSummaryDataPointSlice.At(i)
+// CWMetricMetadata represents the metadata associated with a given CloudWatch metric
+type CWMetricMetadata struct {
+	Namespace                  string
+	TimestampMs                int64
+	LogGroup                   string
+	LogStream                  string
+	InstrumentationLibraryName string
 }
 
 // TranslateOtToCWMetric converts OT metrics to CloudWatch Metric format
 func TranslateOtToCWMetric(rm *pdata.ResourceMetrics, config *Config) ([]*CWMetrics, int) {
 	var cwMetricList []*CWMetrics
-	namespace := config.Namespace
 	var instrumentationLibName string
 
-	if len(namespace) == 0 {
-		serviceName, svcNameOk := rm.Resource().Attributes().Get(conventions.AttributeServiceName)
-		serviceNamespace, svcNsOk := rm.Resource().Attributes().Get(conventions.AttributeServiceNamespace)
-		if svcNameOk && svcNsOk && serviceName.Type() == pdata.AttributeValueSTRING && serviceNamespace.Type() == pdata.AttributeValueSTRING {
-			namespace = fmt.Sprintf("%s/%s", serviceNamespace.StringVal(), serviceName.StringVal())
-		} else if svcNameOk && serviceName.Type() == pdata.AttributeValueSTRING {
-			namespace = serviceName.StringVal()
-		} else if svcNsOk && serviceNamespace.Type() == pdata.AttributeValueSTRING {
-			namespace = serviceNamespace.StringVal()
-		}
-	}
-
-	if len(namespace) == 0 {
-		namespace = defaultNameSpace
-	}
+	cWNamespace := getNamespace(rm, config.Namespace)
+	logGroup, logStream := getLogInfo(rm, cWNamespace, config)
+	timestampMs := time.Now().UnixNano() / int64(time.Millisecond)
```

Comment:
Does `time.Now() / time.Millisecond` work?

Reply:
it looks not, `time.Now()` returns `Time` struct that can't be divided, and it seems `timestampMs := time.Now().UnixNano() / int64(time.Millisecond)` is pretty common practice for getting current timestamp in MS. :)